### PR TITLE
added support for specifying specific branch/tag/revision while installing

### LINKED ISSUE #35: Support for locking onto a Tag or Branch
Lately I'm getting burned by installing Vim plugins at HEAD instead of a stable branch or tag version. It'd be nice to be able to lock onto a stable tag or branch to prevent this from happening. Is this a possibility?

### PR DESCRIPTION
**Pull request by WoLpH**
This fixes #35 

Now it's possible to specify a tag, branch or revision using `@...`:

Example: `Plugin 'WoLpH/nerdtree@patch-1'`

### CODE REVIEW

File: autoload/vundle/installer.vim [hunk 1]
```diff
@@ -393,6 +393,7 @@ func! s:make_sync_command(bang, bundle) abort
                   \ 'git fetch',
                   \ 'git reset --hard origin/HEAD',
```
Comment (luochen1990) [inline autoload/vundle/installer.vim]: `'git reset --hard origin/'.a:bundle.revision,`
I think this is what you want (which was done in line 396 in a wrong way) ?

Comment (WoLpH) [inline autoload/vundle/installer.vim]: That would probably work too. In what way is line 396 wrong? I've been using this fork for a while without any issues

Comment (luochen1990) [inline autoload/vundle/installer.vim]: It's usual that the wrong codes "works", but this doesn't means it's right.

What if the `a:bundle.revision` branch uses a new submodule ?

**Comment by KamranMackey** [comment]
Merging, since this seems like a pretty cool feature.

**Comment by KamranMackey** [comment]
I'll fix it afterwards.

**Comment by gmarik** [comment]
> Merging, since this seems like a pretty cool feature.
@KamranMackey 
1. have anyone other than author tested it? 
2. is it backward compatible?

Before merging require atleast 2 confirmation from other team members

/cc @VundleVim/team

**Comment by KamranMackey** [comment]
@gmarik Oh yeah, crap. Forgot about that. Sorry.

**Comment by gmarik** [comment]
@KamranMackey while it's not too late i'd suggest hard resetting it back to where it was before. Otherwise you'll get flooded with support issues very soon…

**Comment by gmarik** [comment]
@KamranMackey let's figured out process first before getting to merging code. Ok?

**Comment by KamranMackey** [comment]
@gmarik created a PR that reverts #616, since that's the one that might've broken Vundle. I doubt this one did.

**Comment by luochen1990** [comment]
@KamranMackey intersting...

**Comment by WoLpH** [comment]
@gmarik @KamranMackey Perhaps it's time to look at Travis tests? Doesn't have to be fully covering initially but some automated tests might help with accepting pull requests.

**Comment by silfverstrom** [comment]
@WoLpH +1 on adding some automated test-coverage.

**Comment by gmarik** [comment]
:+1: if anyone want's to figure out testing process…

**Comment by gmarik** [comment]
@silfverstrom create separate issues for testing?

**Comment by silfverstrom** [comment]
@gmarik yeah I'll add a new issue.

**Comment by fearphage** [comment]
Is any of this ever coming to master?

**Comment by ryanoasis** [comment]
@fearphage Good question. I tried getting a [next release branch](https://github.com/VundleVim/Vundle.vim/milestone/6) up but we had some issues. I still want to help release the next version but really paranoid about breaking vundle (considering how many vim users use it still).

I have been too busy but hopefully in a few weeks I can contribute more. I have only been merging the most trivial of PRs.

**Pull request merged**